src/common/meta/src/cache/table/composite_table_route.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use futures::future::BoxFuture;
+use moka::future::Cache;
+use snafu::OptionExt;
+use store_api::storage::TableId;
+
+use crate::cache::table::{TableRoute, TableRouteCacheRef};
+use crate::cache::{CacheContainer, Initializer};
+use crate::error;
+use crate::error::Result;
+use crate::instruction::CacheIdent;
+use crate::key::table_route::{LogicalTableRouteValue, PhysicalTableRouteValue};
+
+/// [CompositeTableRoute] stores all level routes of a table.
+/// - Stores [PhysicalTableRouteValue] for logical table.
+/// - Stores [LogicalTableRouteValue], [PhysicalTableRouteValue] for the logical table.
+#[derive(Clone)]
+pub enum CompositeTableRoute {
+    Physical(Arc<PhysicalTableRouteValue>),
+    Logical(Arc<LogicalTableRouteValue>, Arc<PhysicalTableRouteValue>),
+}
```

Comment:
What's the difference between this struct and `TableRoute`, and when to use which?

Reply:
`TableRoute` only contains the first level table route, for logical table, it only contains `LogicalTableRouteValue`. 
We should use the `CompositeTableRoute ` when the system needs to find the `PhysicalTableRouteValue` for a logical table.